### ISSUE TITLE
How to achieve nearly 100 photos per second?

### ISSUE THREAD

**Issue by ericwang321** [AlexeyAB/darknet#4480]
###  :one:. os

ubuntu18.04 64bit + python3
i9 / 32g / RTX 2080ti × 1
alexeyAB / darknet + yolov3

use yolov3.cfg+yolov3.weight,nearly 25 photos per second.

###  :two:.question 

**_Is there any way to achieve_** a processing speed of _**nearly 100 photos per second**_, only **_detecting people_** and counting (**_the original image is 1920 * 1080, dark scene_**) :question:

- reduce the input pic size /

- yolov3-tiny.cfg+retrain sample pic/

- 

**Thanks for any suggestion.**

**Comment by LukeAI** [comment]
100FPS on that hardware should be achievable.

Compile with CUDNN_HALF - should give a big boost.
Use parallel instances of darknet - two or three - I have this in my implementation and it improves GPU utilisation. Unfortunately I can't share the code :/

You could try csresnext50-spp-pan.cfg - bit faster than yolov3-spp, slightly higher accuracy.

Retraining the model to just detect people would make it faster than having the full 80 classes of coco. see the readme for details of how to modify the cfg and extract the pretrained conv weights for transfer learning.

**Comment by ericwang321** [comment]
Thank you for your reply.
CUDNN_HALF  is enabled in Makefile. 
In other words, 
using csresnext50-spp-pan.cfg  and retraining the coco dataset + custom dataset, 
is it possible to reach 50fps?

**Comment by LukeAI** [comment]
csresnext50-spp-pan.cfg will only boost speed ~15% or so.
exploiting parallelism would be the most effective thing to do - maybe double FPS or more.

What image size are you using? how big is your training set? Can you show an example?

**Comment by LukeAI** [comment]
you could also try https://github.com/WongKinYiu/PartialResidualNetworks/files/3629050/darknet53-prn-yolov3.cfg.txt

it supposedly has accuracy almost as high yolov3 but runs twice as fast. Unfortunately pretrained weights are not supplied. so you lose some accuracy there, unless you are prepared to pretrain yourself from scratch on Yolo.

**Comment by LukeAI** [comment]
Please let us know what you try and what the results are, so we can all learn from your work.

**Comment by dreambit** [comment]
@LukeAI Hi

Can u share csresnext50-spp-pan.cfg? Where instructions can be found?
Have u tested it on your dataset and which config do u use on your project?

> exploiting parallelism would be the most effective thing to do - maybe double FPS or more.

Could u without sharing the code explain what do u mean? U use darknet .so and python wrapper?
Or u modified darknet sourse code to run model in 3 instances?

Thanks. :)

**Comment by ericwang321** [comment]
thanks.
width/height  416*416，input size 1920*1080.Prepare 2000 pieces, but the scene is relatively single.
the pic like this but without light.
![5](https://user-images.githubusercontent.com/25045112/70494620-ef058b00-1b46-11ea-9483-64c0a05924b3.png)
I will test and send the results, hope it goes well

**Comment by ericwang321** [comment]
I will temp close this issue,wait test result then reopen.
